firebase 프로젝트 연동

chanho0908 opened the pull request:
## 이슈 번호
<!-- 작업이 완료된 이슈의 번호를 기록해주세요. -->
#15

## 리뷰/머지 희망 기한 (선택)
2025.12.31


## 작업내용
- 크래시틱스/애널리틱스/FCM Gradle 의존성만 간단하게 추가했슴니다 :)

## 결과물
<!-- 구현한 것을 스크린샷, 영상 또는 GIF 형태로 올려주세요. -->

<!---
| Before | After |
|:--:|:--:|
| (이미지) | (이미지) |
!--->


## 리뷰어에게 추가로 요구하는 사항 (선택)
- 애널리틱스 설정을 core:util에 넣을까 고민하다 애널리틱스 설정 및 로깅 관련 코드는 모듈로 분리해서 관리하면
깔끔할 것 같아서 모듈로 분리했는데 어떻게 생각해 ?

chanho0908 committed:
🔥 Firebase 프로젝트 연동 및 서비스 설정

Firebase 프로젝트를 연동하고 Crashlytics, Analytics, Cloud Messaging 서비스를 설정합니다. google-services.json 파일은 .gitignore에 추가하여 버전 관리에서 제외합니다.

🤖 Generated with [Firebender](https://firebender.com)

Co-Authored-By: Firebender <help@firebender.com>

chanho0908 committed:
♻️ Google Services 플러그인을 버전 카탈로그로 마이그레이션

chanho0908 committed:
♻️ 파이어베이스 의존성 수정

chanho0908 committed:
➕ Analytics 모듈 추가 및 Feature 플러그인 통합

Firebase Analytics를 위한 core:analytics 모듈을 생성하고, 모든 Feature 모듈에서 자동으로 사용할 수 있도록 FeatureConventionPlugin에 통합했습니다.

🤖 Generated with [Firebender](https://firebender.com)

Co-Authored-By: Firebender <help@firebender.com>

chanho0908 committed:
➕ Analytics 모듈 추가 및 Feature 플러그인 통합

Firebase Analytics를 위한 core:analytics 모듈을 생성하고, 모든 Feature 모듈에서 자동으로 사용할 수 있도록 FeatureConventionPlugin에 통합했습니다.

🤖 Generated with [Firebender](https://firebender.com)

Co-Authored-By: Firebender <help@firebender.com>

chanho0908 committed:
🙈 .cursorrules를 Git 추적에서 제외

개발 환경별 설정 파일인 .cursorrules를 .gitignore에 추가하여 버전 관리에서 제외합니다.

🤖 Generated with [Firebender](https://firebender.com)

Co-Authored-By: Firebender <help@firebender.com>

dogmania commented on core/analytics/build.gradle.kts:
```diff
@@ -0,0 +1,13 @@
+
+dependencies {
+    implementation(platform(libs.google.firebase.bom))
+    api(libs.google.firebase.analytics)
```
혹시 이 부분만 따로 api로 뺀 이유가 있을까요?? api로 노출시키면 :feature에서 sdk에 접근이 가능해지는데 특별한 이유가 있는 건지 궁금합니다

모듈을 나눈 만큼, 관련된 책임은 이 모듈에서 담당하고 외부 모듈에서 sdk에 직접 접근하는 건 막는 게 좋을 거 같아요. :core:analytics에 로깅 클래스를 구현하고 외부에는 인터페이스를 통해 접근하는 것으로 구현하면 어떨까요? 그럼 api는 사실상 불필요해지고 implementation으로 바꿔도 될 것 같습니다!

chanho0908 replied:
헉 맞아 feature 모듈에서 애널리틱스 의존성이 필요할거라고 예상했었는데
생각해보니 현수말대로 인터페이스로 접근하면 해당 모듈 내부에선 어떤 라이브러리를 쓰는지 몰라도 되겠넹 !

리뷰 반영 커밋 : https://github.com/YAPP-Github/Twix-Android/pull/19/commits/738b438987de5dbd9b308745732a3ea77296148f

dogmania reviewed (commented):
고생하셨습니다! api 부분만 궁금해서 한번 확인 부탁드려용

chanho0908 committed:
♻️ Analytics 모듈의 Firebase 의존성을 api에서 implementation으로 변경

불필요한 의존성 전이를 방지하기 위해 Firebase Analytics 의존성의 가시성을 제한했습니다.

🤖 Generated with [Firebender](https://firebender.com)

Co-Authored-By: Firebender <help@firebender.com>

dogmania reviewed (approved):
고생하셨습니다!
